backend/transaction/index.js

```diff
@@ -1,69 +1,82 @@
-/**
- * Transaction-Service/index.js
- * 
- * - Entry Point: Initializes the Transaction Service.
- * - Purpose: Manages financial transactions related to wallets and stocks.
- * - Server Setup:
- *   - Uses Express.js to handle HTTP requests.
- *   - Connects to MongoDB using Mongoose.
- * - API Endpoints: 
- *   - Fetches wallet transactions (`/api/walletTransactions`).
- *   - Fetches stock transactions (`/api/stockTransactions`).
- * - Modular Design:
- *   - **Controllers:** (`walletController.js`, `stockController.js`) handle business logic.
- *   - **Models:** (`WalletTransaction.js`, `StockTransaction.js`) define database schemas.
- *   - **index.js:** Only routes requests and delegates processing.
- * - Additional Features:
- *   - JSON parsing middleware.
- *   - Health check endpoint (`/`).
- *   - Structured error handling for MongoDB connectivity.
- * - **Scalability:** Designed for easy expansion with new transaction types or financial services.
- */
-
-
-
 const express = require("express");
 const mongoose = require("mongoose");
 const dotenv = require("dotenv");
+const jwt = require("jsonwebtoken"); // ✅ Import `jsonwebtoken` here instead of `authMiddleware.js`
 
-dotenv.config(); // Load environment variables
+dotenv.config();
 
 const app = express();
 const PORT = process.env.PORT || 5000;
-const MONGO_URI = process.env.MONGO_URI || "mongodb://mongo_db:27017/transactionDB";
+const MONGO_URI = process.env.MONGO_URI || "mongodb://localhost:27017/transactionDB";
 
 // Connect to MongoDB
 mongoose
-  .connect(MONGO_URI, { useNewUrlParser: true, useUnifiedTopology: true })
+  .connect(MONGO_URI)
   .then(() => console.log("Transaction-Service/index.js: ✅ MongoDB Connected Successfully"))
-  .catch((err) => console.error("Transaction-Service/index.js: ❌ MongoDB Connection Error:", err));
+  .catch((err) => {
+    console.error("❌ MongoDB Connection Error:", err);
+    process.exit(1); // Exit on failure to prevent silent errors
+  });
 
 app.use(express.json()); // Middleware to parse JSON requests
 
-// Import Controllers & Models (No Routes, Calls Controllers Directly)
+// ✅ Inline Authentication Middleware (Replaces `authMiddleware.js`)
+const authMiddleware = (req, res, next) => {
```

Comment:
Why we implemented a new authmiddleware for this?

Reply:
spent 7 hours trying to import the middleware and getiing it to waore and ir was not working --> figured it was best to jsut get it done